apps/OboeTester/app/src/main/cpp/SawPingGenerator.cpp

```diff
@@ -28,18 +29,16 @@ SawPingGenerator::SawPingGenerator()
 
 SawPingGenerator::~SawPingGenerator() { }
 
-
-AudioResult SawPingGenerator::onProcess(
-        uint64_t framePosition,
+int32_t SawPingGenerator::onProcess(
+        int64_t framePosition,
         int numFrames) {
 
     frequency.pullData(framePosition, numFrames);
     amplitude.pullData(framePosition, numFrames);
 
-    const float *frequencies = frequency.getFloatBuffer(numFrames);
-    const float *amplitudes = amplitude.getFloatBuffer(numFrames);
-    float *buffer = output.getFloatBuffer(numFrames);
-
+    const float *frequencies = frequency.getBlock();
```

Comment:
Passing around raw pointers allows the caller to delete the underlying object. As discussed, let's investigate options for using smart pointers and/or providing safer access to the block/buffer.